pubsub/streaming-analytics/README.md

```diff
@@ -0,0 +1,150 @@
+# Stream Cloud Pub/Sub with Cloud DataFlow
+
+Samples showing how to use [Google Cloud Pub/Sub] with [Google Cloud Dataflow].
+
+## Before you begin
+
+1. Install the [Cloud SDK].
+
+1. [Create a new project].
+
+1. [Enable billing].
+
+1. [Enable the APIs](https://console.cloud.google.com/flows/enableapi?apiid=dataflow,compute_component,logging,storage_component,storage_api,pubsub,cloudresourcemanager.googleapis.com,cloudscheduler.googleapis.com): Dataflow, Compute Engine, Stackdriver Logging, Cloud Storage, Cloud Storage JSON, Pub/Sub, Cloud Scheduler, and Cloud Resource Manager.
+
+1. Setup the Cloud SDK to your GCP project.
+
+   ```bash
+   gcloud init
+   ```
+
+1. [Create a service account key] as a JSON file.
+   For more information, see [Creating and managing service accounts].
+
+   * From the **Service account** list, select **New service account**.
+   * In the **Service account name** field, enter a name.
+   * From the **Role** list, select **Project > Owner**.
+
+     > **Note**: The **Role** field authorizes your service account to access resources.
+     > You can view and change this field later by using the [GCP Console IAM page].
+     > If you are developing a production app, specify more granular permissions than **Project > Owner**.
+     > For more information, see [Granting roles to service accounts].
+
+   * Click **Create**. A JSON file that contains your key downloads to your computer.
+
+1. Set your `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point to your service account key file.
+
+   ```bash
+   export GOOGLE_APPLICATION_CREDENTIALS=path/to/your/credentials.json
+   ```
+
+1. Create a Cloud Storage bucket. Grant the Default Compute Engine Service Account Object Admin Access to your bucket.
+
+   ```bash
+   gsutil mb gs://your-gcs-bucket
+   
+   BUCKET=gs://your-gcs-bucket
+   PROJECT=${gcloud config get-value project}
+   PROJECT_NUMBER=$(gcloud projects describe ${PROJECT} --format="value(projectNumber)")
+   
+   gsutil iam ch \
```

Comment:
By the way, on a new project by following these instructions, I was able to run everything without having to do this `gsutil iam ch` both in the DirectRunner as well as in DataflowRunner.

Reply:
What about for an existing project that was created a long time ago? I'd like to keep this as optional in case people run into issues like I did.

Reply:
I removed it. Thanks for verifying!